core/ir/x86/decode_private.h

```diff
@@ -144,15 +144,17 @@ enum {
  * 2nd = 2nd byte of opcode (if there are 2)
  * 3rd (ls) = split into nibbles
  *   1st nibble (ms) = if bit 1 (OPCODE_TWOBYTES) set, opcode has 2 bytes
+ *                       if REQUIRES_EVEX then this bit instead means that this
+ *                       instruction must have evex.b set
```

Comment:
I would expect this to instead be a flag like REQUIRES_VEX_L0?

In fact I'm not remembering why we have what seems a hacky overload of these opcode bits for the vex.W, etc. below: do we not have access to the flags field when we need that info?

Reply:
Those flags are only used for decoding, not encoding.

Reply:
Hmm, it seems like we could use them for encoding, right?  Maybe at least add a comment about switching to that in the future, which seems a little cleaner.